devito/types/sparse.py

```diff
@@ -742,7 +745,11 @@ def __shape_setup__(cls, **kwargs):
                 raise ValueError('`nt` must be > 0')
 
             shape = list(AbstractSparseFunction.__shape_setup__(**kwargs))
-            shape.insert(cls._time_position, nt)
+            try:
```

Comment:
using assert for control flow is unpythonic (asserts can be stripped with -O and should signal programmer errors, not drive logic).

Reply:
also, in the unlucky circumstance that `nt == x_size (+ halo etc)` we would have an issue here no?

I think we need to check `len(shape)` instead...

Or even better, ensure we don't need to worry about this at this point, which seems a bit weird